File: pythonCode/pract04.py
#-------------------------------------------------------------------------------
# Andrew Stanton
# 683693
# Autumn Teaching Block
#-------------------------------------------------------------------------------
from graphics import *
def personalGreeting():
    Name = input("What is your name: ")
    Name = Name.capitalize()
    strOut =  "Hello {0}, nice to see you.".format(Name)
    print(strOut)

def formalName():
    FirstName = input("First Name: ").capitalize()
    LastName = input("Last Name: ").capitalize()
    strOut = "{0}. {1}".format(FirstName[0], LastName)
    print(strOut)

def kilos2pounds():
    kilos = eval(input("Enter a weight in kilograms: "))
    pounds = 2.2 * kilos
    print("The weight in pounds is {0:0.2f}".format(pounds))

def generateEmail():
    FirstName = input("First Name: ").lower()
    LastName = input("Last Name: ").lower()
    Year = input("Year of entry")
    strOut ="{0}.{1}.{2}@myport.ac.uk".format(LastName[:4], FirstName[0],
                                                Year[2:])
    print(strOut)

def gradeTest():
    Grades = "FFFFCCBBAAA"
    Score = int(eval(input("Score: ")))
    print(Grades[Score])

def graphicLetters():
    Word = input("Give me a word: ")
    win = GraphWin()
    for ch in Word:
        p = win.getMouse()
        Label = Text(p, ch)
        Label.setSize(20)
        Label.draw(win)

def singASong():
    Lyric = input("Lyric: ")
    Lines = int(input("Lines: "))
    LyricsPerLine = int(input("How many lyrics per line: "))
    if Lyric[-1] != " ":
        Lyric += " "
    for i in range(Lines):
        print(Lyric*LyricsPerLine)

def exchangeTable():
    print("{0}\t{1}".format("Euros", "Pounds"))
    for i in range(0, 21):
        print("{0:5}\t{1:5.2f}".format(i, i/1.15))

def makeAcronym():
    strIn = input("Word to generate acronym for: ")
    strList = strIn.split()
    for word in strList:
        print(word[0].capitalize() , end="")

def nameToNumber():
    strIn = input("Name: ")
    total = 0
    for ch in strIn:
        total += ord(ch)
    print("{0} has the value of {1}".format(strIn, total))

def fileInCaps():
    filename = input("Filename: ")
    file = open(filename, "r")
    content = file.read()
    content = content.upper()
    print(content)
    file.close()

def rainfallChart():
    filestream = open("rainfall.txt", "r")
    for lines in filestream:
        data = lines.split()
        strOut = "{0:<14}{1}".format(data[0]+":", "*"*int(data[1]))
        print(strOut)
    filestream.close()

def rainfallInInches():
    filestreamIn = open("rainfall.txt", "r")
    theData = filestreamIn.readlines()
    filestreamIn.close()
    filestreamOut = open("rainfallInches.txt", "w")
    for lines in theData:
        data = lines.split()
        strOut = "{0:<14}{1:0.2f}".format(data[0], float(data[1])/25.4)
        print(strOut, file=filestreamOut)
    filestreamOut.close()

def wc():
    filePath = input("Name of file: ")
    filestream = open(filePath, "r")
    data = filestream.readlines()
    print("Number of lines: {0}".format(len(data)))
    noOfWords, noOfChars = 0, 0
    for lines in data:
        noOfChars += len(lines)
        noOfWords += len(lines.split())
    print("Number of words: {0}".format(noOfWords))
    print("Number of characters: {0}".format(noOfChars))
    print(data[0])
    print(len(data[0]))

File: pract04.pl
use strict;
use warnings;
use 5.010;  # For say
use input;
sub personalGreeting{
    print 'What is your name: ';
    my $name = <>;
    chomp $name;    # Removes $/ (Input record separator) Currently new line. != chop
    print "Hello $name, nice to see you.\n";    # variable interpolation means sprintf not needed.
}
sub formalName{
    print "First name: ";
    my $firstName = <>;
    my $initial = uc substr $firstName, 0, 1;
    print "Last name: ";
    my $lastName = ucfirst lc <>;
    print "$initial.$lastName";
}
sub kilos2Pounds{
    my $kilos = input::int('Enter a weight in kilograms');
    my $pounds = 2.2 * $kilos;
    printf "%.2f kilos is equal to %.2f pounds\n", ($kilos, $pounds);
}
sub generateEmail{
    print "Enter first name: ";
    my $firstName = <>;
    print "Enter last name: ";
    my $lastName = <>;
    my $year = input::int("Enter year of enrolment");
    printf "%.4s.%.1s.%s\@myport.ac.uk\n", ($lastName, $firstName, substr($year, -2));
}
sub gradeTest{
    my $grades = 'FFFFCCBBAAA';
    my $mark = input::int("Enter mark");
    if (1 <= $mark && $mark <= 10 ){
        my $result = substr $grades, $mark, 1;
        print "$result\n";
    }
    else{
        print "Enter a value in range 0 .. 10\n";
    }
}
sub singASong{
    print 'Enter lyric: ';
    my $lyric = <>;
    chomp $lyric;
    my $lines = input::int('Enter number of lines');
    my $lyricsPerLine = input::int('Enter number of lyrics per line');
    my $line = ($lyric.' ') x $lyricsPerLine;
    $line .= "\n";
    foreach(1 .. $lines) {
        print $line;
    }
}
sub exchangeTable{
    print "Euros\tPounds\n";
    foreach(0 .. 20){
        printf "%3u\t%5.2f\n", ($_, $_/1.15);
    }
}
sub makeAcronym{
    print "Enter sentence to make acronym: ";
    my $input = <>;
    my @words = split / /, $input;
    foreach(@words) {
        print uc substr $_, 0, 1;
    }
    print "\n";
}
sub nameToNumber{
    my $total = 0;
    print "Enter your name: ";
    my $name = <>;
    chomp $name;
    $name = uc $name;
    my $wordLength = length $name;
    foreach(0 .. $wordLength - 1) {
        my $char = substr $name, $_, 1;
        my $numeric = (ord $char) - 64;
        $total += $numeric;
    }
    say "$name = $total";
}
sub filesInCaps{
    print "Enter filename: ";
    my $fileName = <>;
    open ('file', '<'.$fileName) or die 'Can\'t open file';
    while (<file>){
        print uc;
    }
}
sub rainfallChart{
    open ('data', '<rainfall.txt') or die 'Can\'t open file';
    while(<data>){
        my ($location, $rain) = split;
        printf "%13s||%s\n", ($location, '*' x $rain);
    }
    close 'data';
}
sub rainfallInInches{
    open ('rainIn', '<rainfall.txt') or die 'Can\'t open file';
    open (my $rainOut, '>', 'rainfallInches.txt') or die 'Can\'t open file';
    while(<rainIn>){
        my ($location, $rain) = split;
        my $rainInches = $rain / 25.4;
        printf <$rainOut>, '%s %.2f\n', ($location, $rainInches);
    }
    close 'rainIn';
    close $rainOut;
}
rainfallInInches();
